mir-ci/mir_ci/conftest.py

```diff
@@ -225,3 +225,23 @@ def xdg(request: pytest.FixtureRequest, tmp_path: pathlib.Path) -> Generator:
                     f.write(contents)
             m.setenv(var, str(var_path))
         yield
+
+
+@pytest.fixture(scope="function")
+def env(request: pytest.FixtureRequest) -> Generator:
+    """
+    Set environment variables.
+
+    Use the `env` mark to provide data, e.g.:
+    @pytest.mark.env(ENV_VARIABLE='value')
+    """
+    if request.config.getoption("--deps"):
+        yield
+        return
+
+    with pytest.MonkeyPatch.context() as m:
+        marks: Iterator[pytest.Mark] = request.node.iter_markers("env")
+        for mark in marks:
+            for var, value in mark.kwargs.items():
```

Comment:
You should reverse this, as `.iter_markers` will walk up the stack to find the marks, and you need the closest mark to override any previous ones.

Reply:
Done.